src/D2L.CodeStyle.Analyzers/Immutability/ImmutabilityContext.cs

```diff
@@ -182,6 +182,14 @@ out diagnostic
 				return true;
 			}
 
+			if( type.IsTupleType ) {
+				info = ImmutableTypeInfo.CreateWithAllImmutableTypeParameters(
+					kind: ImmutableTypeKind.Total,
+					type: type

```

Comment:
Is it "weird" that we don't do `.OriginalDefinition` here? It works regardless, though?

Reply:
Doesn't really matter. This is basically JIT adding `[Immutable]` attributes to the type definition.

Reply:
Yeah, it's just interesting to me that type could be like, `ValueType<object>`, no?

Reply:
`[Immutable]ValueType<[Immutable]T1>`, `[Immutable]ValueType<[Immutable]T1, [Immutable]T2>`... are things I'd expect to express with an `ImmutableTypeInfo` for sure.

Reply:
I can switch it to OrignialDefinition, I've no strong feels.

```csharp
			readonly (ClassMarkedImmutableImplementingRegularInterface) m_bar = default;
			readonly (object) /* NonImmutableTypeHeldByImmutable(Class, Object, ) */ m_foo /**/ = default;
```

```
type.Name
"ValueTuple"
type.TypeParameters
Length = 2
    [0]: {T1}
    [1]: {T2}
    Results View: Expanding the Results View will enumerate the IEnumerable
type.TypeArguments
Length = 2
    [0]: {SpecTests.Classes.ClassMarkedImmutableImplementingRegularInterface}
    [1]: {?}
    Results View: Expanding the Results View will enumerate the IEnumerable
type.Name
"ValueTuple"
type.TypeParameters
Length = 2
    [0]: {T1}
    [1]: {T2}
    Results View: Expanding the Results View will enumerate the IEnumerable
type.TypeArguments
Length = 2
    [0]: {object}
    [1]: {?}
    Results View: Expanding the Results View will enumerate the IEnumerable
type.ToMinimalDisplayString()
error CS1501: No overload for method 'ToMinimalDisplayString' takes 0 arguments
type.ToDisplayString()
"(object, ?)"
type.OriginalDefinition.ToDisplayString()
"(T1, T2)"
```

Reply:
The `?` (ErrorType) is because I suppose single-valued tuples aren't valid, need a minimum of two :P

Still works though!

Reply:
Yeah, I don't feel strongly about it and suspected it would still work. I'm just "surprised" to see it not used, even though in practice it doesn't matter.

It might matter more if we want to support what `[ImmutableGeneric]` is doing, where we say "I know `IFactory<T>` in general is mutable, but I want any `IFactory<Foo>` to be immutable". I don't think the logic of that stuff is 100% sound though, like when you have `Bar<X>` implementing `IFactory<X>` then `Bar<Foo>` would also need to be immutable too and I'm not sure we currently do that right (so I'm happy to stop supporting that feature)